core/src/main/java/com/sequenceiq/cloudbreak/service/upgrade/sync/operationresult/CmSyncOperationResultEvaluatorService.java

```diff
@@ -0,0 +1,65 @@
+package com.sequenceiq.cloudbreak.service.upgrade.sync.operationresult;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.google.common.collect.Sets;
+import com.sequenceiq.cloudbreak.service.upgrade.sync.common.ParcelInfo;
+
+@Service
+public class CmSyncOperationResultEvaluatorService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CmSyncOperationResultEvaluatorService.class);
+
+    public CmSyncOperationSummary.Builder evaluateParcelSync(CmParcelSyncOperationResult cmParcelSyncOperationResult) {
+        if (cmParcelSyncOperationResult.getInstalledParcels().isEmpty()) {
+            String message = "CM parcel sync failed, it was not possible to retrieve installed parcel versions from the CM server.";
+            LOGGER.debug(message);
+            return CmSyncOperationSummary.builder().withError(message);
+        }
+
+        Set<String> installedParcelNames = cmParcelSyncOperationResult.getInstalledParcels().stream().map(ParcelInfo::getName).collect(Collectors.toSet());
+        Set<String> foundCmProductNames = cmParcelSyncOperationResult.getFoundCmProducts().stream().map(cmp -> cmp.getName()).collect(Collectors.toSet());
+        Set<String> notFoundProductNames = Sets.difference(installedParcelNames, foundCmProductNames);
+        if (notFoundProductNames.isEmpty()) {
+            String message = String.format("Successfully synced following parcel versions from CM server: %s", installedParcelNames);
+            LOGGER.debug(message);
+            return CmSyncOperationSummary.builder().withSuccess(message);
+        } else {
+            String message = getParcelErrorMessage(installedParcelNames, foundCmProductNames, notFoundProductNames);
+            LOGGER.debug(message);
+            return CmSyncOperationSummary.builder().withError(message);
+        }
+    }
+
+    public CmSyncOperationSummary.Builder evaluateCmRepoSync(CmRepoSyncOperationResult cmRepoSyncOperationResult) {
+        if (cmRepoSyncOperationResult.getInstalledCmVersion().isEmpty()) {
+            String message = "CM repository sync failed, it was not possible to retrieve CM version from the server.";
+            LOGGER.debug(message);
+            return CmSyncOperationSummary.builder().withError(message);
+        }
```

Comment:
Hm. I can do but this is a return early pattern. I do not think that it does need an else. But if you insist, I can add.
I move the rest out to a separate method.

Reply:
Ok, I added the else, I think that is a more often used pattern in this repo.

Reply:
I don't like the "early return" pattern because it conflicts with readability. Anyway, your code with an else would still exit the same place but it would be more structured. Also the empty line indicated that this should be refactored